[Key Vault Secrets] Can't compile samples on Release mode

Repro:
```
cmake -DBUILD_TESTING=ON -DBUILD_SAMPLES=ON -DCMAKE_BUILD_TYPE=Release ..
cmake --build . --target sample2-backup-restore sample3-delete-recover 
```

build exception:
```
FAILED: sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/CMakeFiles/sample3-delete-recover.dir/sample3_delete_recover.cpp.o 
[build] /usr/bin/clang++-10 -DAZ_RTTI -DBUILD_CURL_HTTP_TRANSPORT_ADAPTER -DCURL_STATICLIB -DTESTING_BUILD -I/home/vivazqu/workspace/azure-sdk-for-cpp/sdk/keyvault/azure-security-keyvault-secrets/inc -I/home/vivazqu/workspace/azure-sdk-for-cpp/sdk/core/azure-core/inc -I/home/vivazqu/workspace/azure-sdk-for-cpp/sdk/identity/azure-identity/inc -isystem /home/vivazqu/workspace/vcpkg/installed/x64-linux/include -O3 -DNDEBUG -fno-operator-names -Wold-style-cast -Xclang -Wall -Wextra -pedantic -Werror -Wdocumentation -Wdocumentation-unknown-command -Wcast-qual -pthread -std=gnu++14 -MD -MT sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/CMakeFiles/sample3-delete-recover.dir/sample3_delete_recover.cpp.o -MF sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/CMakeFiles/sample3-delete-recover.dir/sample3_delete_recover.cpp.o.d -o sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/CMakeFiles/sample3-delete-recover.dir/sample3_delete_recover.cpp.o -c /home/vivazqu/workspace/azure-sdk-for-cpp/sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/sample3_delete_recover.cpp
[build] /home/vivazqu/workspace/azure-sdk-for-cpp/sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/sample3_delete_recover.cpp:106:3: error: use of undeclared identifier 'assert'
[build]   assert(expected.Name == actual.Name);
[build]   ^
[build] /home/vivazqu/workspace/azure-sdk-for-cpp/sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/sample3_delete_recover.cpp:107:3: error: use of undeclared identifier 'assert'
[build]   assert(expected.Properties.Version == actual.Properties.Version);
[build]   ^
[build] /home/vivazqu/workspace/azure-sdk-for-cpp/sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/sample3_delete_recover.cpp:108:3: error: use of undeclared identifier 'assert'
[build]   assert(expected.Id == actual.Id);
[build]   ^
[build] 3 errors generated.
```

The function `AssertSecretsEqual` from the sample is non-compilable on Release mode.


Make sure to add `build samples in Release mode in CI` after fixing this

--- a/CMakeSettings.json
+++ b/CMakeSettings.json
@@ -170,6 +170,11 @@
           "name": "BUILD_PERFORMANCE_TESTS",
           "value": "True",
           "type": "BOOL"
+        },
+        {
+          "name": "BUILD_SAMPLES",
+          "value": "True",
+          "type": "BOOL"
         }
       ]
     },

--- a/eng/pipelines/templates/jobs/archetype-sdk-tests.yml
+++ b/eng/pipelines/templates/jobs/archetype-sdk-tests.yml
@@ -59,7 +59,7 @@ jobs:
         OSVmImage: MMSUbuntu18.04
         VcpkgInstall: 'curl[ssl] libxml2 openssl'
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
-        CmakeArgs: ' -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Release'
+        CmakeArgs: ' -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_SAMPLES=ON'
         AZURE_CORE_ENABLE_JSON_TESTS: 1
         BuildArgs: '-j 4'
       Ubuntu20_samples:
@@ -95,7 +95,7 @@ jobs:
         VCPKG_DEFAULT_TRIPLET: 'x86-windows-static'
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: Win32
-        CmakeArgs: ' -DBUILD_RTTI=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=ON'
+        CmakeArgs: ' -DBUILD_RTTI=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=ON -DBUILD_SAMPLES=ON'
         BuildArgs: '-v --parallel 8'
       Win_x64_with_unit_test_winHttp:
         Pool: azsdk-pool-mms-win-2019-general
@@ -103,7 +103,7 @@ jobs:
         VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: x64
-        CmakeArgs: ' -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON '
+        CmakeArgs: ' -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DBUILD_SAMPLES=ON'
         BuildArgs: '-v --parallel 8'
         AZURE_CORE_ENABLE_JSON_TESTS: 1
       # specify libcurl to be used on Windows
@@ -132,7 +132,7 @@ jobs:
         OSVmImage: 'macOS-10.15'
         VcpkgInstall: 'curl[ssl] libxml2 openssl'
         VCPKG_DEFAULT_TRIPLET: 'x64-osx'
-        CmakeArgs: ' -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON '
+        CmakeArgs: ' -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DBUILD_SAMPLES=ON'
         AZURE_CORE_ENABLE_JSON_TESTS: 1
         BuildArgs: '-j 4'
   pool:

--- a/sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample2-backup-restore/sample2_backup_restore.cpp
+++ b/sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample2-backup-restore/sample2_backup_restore.cpp
@@ -20,6 +20,7 @@
 #include <azure/identity.hpp>
 #include <azure/keyvault/keyvault_secrets.hpp>
 
+#include <assert.h>
 #include <chrono>
 #include <fstream>
 #include <iostream>

--- a/sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/sample3_delete_recover.cpp
+++ b/sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample3-delete-recover/sample3_delete_recover.cpp
@@ -20,9 +20,9 @@
 #include <azure/identity.hpp>
 #include <azure/keyvault/keyvault_secrets.hpp>
 
+#include <assert.h>
 #include <chrono>
 #include <iostream>
-
 using namespace Azure::Security::KeyVault::Secrets;
 using namespace std::chrono_literals;
 void AssertSecretsEqual(KeyVaultSecret const& expected, KeyVaultSecret const& actual);

